src/containers/my-cooperations/empty-cooperation-activities/EmptyCooperationActivities.tsx

```diff
@@ -1,118 +1,59 @@
+import { ReactElement } from 'react'
 
 const EmptyCooperationActivities = () => {
+  const { userRole } = useAppSelector((state) => state.appMain)
   const { t } = useTranslation()
-  const { openModal, closeModal } = useModalContext()
-  const { setIsActivityCreated } = useCooperationContext()
-  const { openMenu, renderMenu, closeMenu } = useMenu()
-  const [isMenuOpen, setIsMenuOpen] = useState<boolean>(false)
-
-  const handleMenuClick = () => (event: MouseEvent<HTMLButtonElement>) => {
-    openMenu(event)
-    setIsMenuOpen((prevState) => !prevState)
-  }
-
-  const openAddCourseTemplateModal = () => {
-    closeMenu()
-    openModal({
-      component: <AddCourseTemplateModal closeModal={closeModal} />
-    })
-  }
-
-  const menuIcon = isMenuOpen ? (
-    <KeyboardArrowUpIcon />
-  ) : (
-    <KeyboardArrowDownIcon />
-  )
-
-  const handleFromScratch = () => {
-    closeMenu()
-    setIsActivityCreated(true)
-  }
 
-  const menuItems = [
-    {
-      id: 1,
-      label: <>{t('cooperationsPage.manyTypes.courseTemplate')}</>,
-      icon: <ViewComfyOutlinedIcon />,
-      onClick: openAddCourseTemplateModal
-    },
-    {
-      id: 2,
-      label: <>{t('cooperationsPage.manyTypes.scratch')}</>,
-      icon: <Crop75Icon />,
-      onClick: handleFromScratch
-    }
-  ]
-
-  const menu = menuItems.map(({ id, label, icon, onClick }) => (
-    <MenuItem key={id} onClick={onClick} sx={styles.menuItem}>
-      {icon} {label}
-    </MenuItem>
-  ))
-
-  const createDescriptionSpan = (text: string) => (
+  const createDescriptionSpan = ({
+    text,
+    context,
+    from = 'description'
+  }: {
+    text: string
+    context?: string
+    from?: string
+  }) => (
     <Typography component={ComponentEnum.Span} sx={styles.weightBox}>
-      {t(`cooperationsPage.description.${text}`)}
+      {t(`cooperationsPage.${from}.${text}`, { context })}
     </Typography>
   )
 
-  const componentDescription = (
-    <>
-      {t('cooperationsPage.description.existingCourse')}
-      {createDescriptionSpan('courseTemplate')}
-      {t('cooperationsPage.description.resourceLibrary')}
-      {createDescriptionSpan('module')}
-      {t('cooperationsPage.description.fillThis')}
-      {createDescriptionSpan('lessons')}
-      {t('cooperationsPage.description.or')}
-      {createDescriptionSpan('quizzes')}
-      {t('cooperationsPage.description.resourcesLibrary')}
-    </>
-  )
-
-  const startIcon = <AddIcon fontSize={SizeEnum.Small} />
-
-  const menuCondition = renderMenu(menu)
+  const textDescription = () => {
+    let description: string | ReactElement = ''
+
+    if (userRole) {
+      description = (
+        <>
+          {componentDescription[userRole]?.map(({ text, isSpan }) =>
+            isSpan
+              ? createDescriptionSpan({ text })
+              : t(`cooperationsPage.description.${text}`)
+          )}
+        </>
+      )
+    }
+    return description
+  }
```

Comment:
I think creating a function for this use case in redundant as well as using fragment. I suggest refactoring like following:

```suggestion
  
let description: ReactNode = "";

if (userRole in componentDescription) {
  description = componentDescription[userRole].map(({ text, isSpan }) =>
    isSpan
      ? createDescriptionSpan({ text })
      : t(`cooperationsPage.description.${text}`)
  );
}
  
```

I use `ReactNode` here because `ReactNode` already includes `string` type in it.